instrumentation/mybatis-3.2/javaagent/src/test/java/io/opentelemetry/javaagent/instrumentation/mybatis/v3_2/MyBatisTest.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.javaagent.instrumentation.mybatis.v3_2;
+
+import io.opentelemetry.api.trace.SpanKind;
+import io.opentelemetry.instrumentation.testing.junit.AgentInstrumentationExtension;
+import io.opentelemetry.instrumentation.testing.junit.InstrumentationExtension;
+import org.apache.ibatis.mapping.Environment;
+import org.apache.ibatis.session.Configuration;
+import org.apache.ibatis.session.SqlSession;
+import org.apache.ibatis.session.SqlSessionFactoryBuilder;
+import org.apache.ibatis.transaction.jdbc.JdbcTransactionFactory;
+import org.h2.jdbcx.JdbcDataSource;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.RegisterExtension;
+
+class MyBatisTest {
+
+  @RegisterExtension
+  protected static final InstrumentationExtension testing = AgentInstrumentationExtension.create();
+
+  private static final String SPAN_NAME =
```

Comment:
As this is used in only 1 place might as well inline the value there.
I'm not sure whether it is deliberate or a coincidence that you have class+method name as the span name. If it is deliberate then our convention for spans based on a code location is to use the class name without the package and use the `code.namespace` (this will have the class name with the package) and `code.function` attributes. Have a look at https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/main/instrumentation/external-annotations/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/extannotations/ExternalAnnotationSingletons.java You can use `CodeSpanNameExtractor.create(codeAttributesGetter))` to get the span name extractor.

Reply:
`code.namespace` and `code.function` attributes are described in https://github.com/open-telemetry/semantic-conventions/blob/d21135e07ce9ca1ba05581021aebe84cc1d27ee8/docs/general/attributes.md#source-code-attributes.
Our convention for spans that represent a method is to use `simple class name.method name` as span name and fill `code.namespace` and `code.function` attributes. Following these conventions and using common code for it allows us to produce uniform telemetry.
Looking at https://github.com/mybatis/mybatis-3/blob/5f11d7dffd6b6625167553b9e46f1244f0d7f8f5/src/main/java/org/apache/ibatis/binding/MapperMethod.java#L222-L240 the name can be either `null` (which current code does not seem to handle) or `MappedStatement#getId()`. As far as I can tell the `MappedStatement#getId()` can also come form xml configuration. Are you sure that doing the substring magic to get rid of the package name would be correct in that case?

Reply:
> `code.namespace` and `code.function` attributes are described in https://github.com/open-telemetry/semantic-conventions/blob/d21135e07ce9ca1ba05581021aebe84cc1d27ee8/docs/general/attributes.md#source-code-attributes. Our convention for spans that represent a method is to use `simple class name.method name` as span name and fill `code.namespace` and `code.function` attributes. Following these conventions and using common code for it allows us to produce uniform telemetry. Looking at https://github.com/mybatis/mybatis-3/blob/5f11d7dffd6b6625167553b9e46f1244f0d7f8f5/src/main/java/org/apache/ibatis/binding/MapperMethod.java#L222-L240 the name can be either `null` (which current code does not seem to handle) or `MappedStatement#getId()`. As far as I can tell the `MappedStatement#getId()` can also come form xml configuration. Are you sure that doing the substring magic to get rid of the package name would be correct in that case?

Thank your explanation for the first question. For the second problem, whether it is in the form of a xml file or annotations like test case provided now, as long as the configuration is correct, relevant information can be effectively obtained in this way. I tested that if it is null, an error will be thrown and stop executing downward before the method getting the span name is executed.
![image](https://github.com/open-telemetry/opentelemetry-java-instrumentation/assets/25423978/07956524-6480-4aa3-92dd-3495da93f155)